src/hooks.ts

```diff
@@ -31,17 +32,20 @@ type PostMultipleWithError<T> = (error: Error, result: DocumentType<T>[], next:
 type PostRegExpWithError<T> = (error: Error, result: NumberOrDocumentOrDocumentArray<T>, next: HookNextErrorFn) => ReturnVoid;
 type PostArrayWithError<T> = (error: Error, result: NumberOrDocumentOrDocumentArray<T>, next: EmptyVoidFn) => ReturnVoid;
 
+type AggregateMethod = 'aggregate';
 type DocumentMethod = 'init' | 'validate' | 'save' | 'remove';
 type NumberMethod = 'count';
 type SingleMethod = 'findOne' | 'findOneAndRemove' | 'findOneAndUpdate' | 'findOneAndDelete' | 'deleteOne' | DocumentMethod;
-type MultipleMethod = 'find' | 'update' | 'deleteMany';
+type MultipleMethod = 'find' | 'update' | 'deleteMany' | 'aggregate';
 type QueryMethod = 'count' | 'countDocuments' | 'estimatedDocumentCount' | 'find' | 'findOne' | 'findOneAndRemove' | 'findOneAndUpdate' | 'update' | 'updateOne' | 'updateMany' | 'findOneAndDelete' | 'deleteOne' | 'deleteMany';
 type ModelMethod = 'insertMany';
 type QMR = QueryMethod | ModelMethod | RegExp;
 type QDM = QMR | DocumentMethod;
 type DR = DocumentMethod | RegExp;
 
 interface Hooks {
+  pre<T>(method: AggregateMethod, fn: PreFnWithAggregate<T>): ClassDecorator;
+
   pre<T>(method: DR | DR[], fn: PreFnWithDocumentType<T>): ClassDecorator;
 
   pre<T>(method: QMR | QMR[], fn: PreFnWithQuery<T>): ClassDecorator;
```

Comment:
is there no `post` hook for `aggregate`?

Reply:
Aggregate return an array I added aggregate to MultipleMethod that have post hooks.

Reply:
right, forgot